config/config.go

```diff
@@ -67,6 +68,17 @@ func Load() (*Config, error) {
 	if c.Server6 == nil && c.Server4 == nil {
 		return nil, ConfigErrorFromString("need at least one valid config for DHCPv6 or DHCPv4")
 	}
+
+	c.v.WatchConfig()
+	c.v.OnConfigChange(func(e fsnotify.Event) {
+		if err := c.parseConfig(protocolV6); err != nil {
```

Comment:
I wonder if this is thread safe since the config is being read from several goroutines. Maybe we should use something like https://golang.org/pkg/sync/atomic/#SwapPointer.

Reply:
You're right this should be tested. I will test it and make adjustments

Reply:
Testing confirms that there is a data race on config update, but only if separate gorutines are writting in the config file, that is also pointed in the issue https://github.com/spf13/viper/issues/174. Reading is not affected by that. After updating config, gorutines sync with the new config at the same time

Reply:
Shouldn't this be done inside `parseConfig`? https://github.com/coredhcp/coredhcp/blob/master/config/config.go#L168

Reply:
You're right. I pushed a fix and iplemented pointer swapping for both v4 and v6